code/starfield/starfield.cpp

```diff
@@ -563,34 +569,104 @@ void parse_startbl(const char *filename)
+
 			// normal debris pieces
-			while (optional_string("$Debris:")) {
+			// leaving this for retail - Mjn
+			if (check_for_string("$Debris:")) {
+
+				mprintf(("Using deprecated motion debris parsing for Default motion debris!\n"));
+				motion_debris this_debris;
+				this_debris.name = "Default";
+
+				int count = 0;
+
+				while (optional_string("$Debris:")) {
 				in_check = true;
 
-				stuff_string(name, F_NAME, MAX_FILENAME_LEN);
+					stuff_string(name, F_NAME, MAX_FILENAME_LEN);
 
-				if (Num_debris_normal < MAX_DEBRIS_VCLIPS) {
-					strcpy_s(Debris_vclips_normal[Num_debris_normal++].name, name);
+					if (count < MAX_MOTION_DEBRIS_BITMAPS) {
+						strcpy_s(this_debris.bitmaps[count++].name, name);
+					} else {
+						Warning(LOCATION, "Could not load normal motion debris '%s'; maximum of %d exceeded.", name, MAX_MOTION_DEBRIS_BITMAPS);
+					}
 				}
-				else {
-					Warning(LOCATION, "Could not load normal motion debris '%s'; maximum of %d exceeded.", name, MAX_DEBRIS_VCLIPS);
+				if (count == MAX_MOTION_DEBRIS_BITMAPS) {
+					Motion_debris.push_back(this_debris);
+				} else {
+					error_display(0, "Not enough bitmaps defined for motion debris '%s'. Skipping!\n", this_debris.name.c_str());
 				}
 			}
 
 			CHECK_END();
 
 			// nebula debris pieces
-			while (optional_string("$DebrisNeb:")) {
+			// leaving this for retail - Mjn
+			if (check_for_string("$DebrisNeb:")) {
+
+				mprintf(("Using deprecated motion debris parsing for Nebula motion debris!\n"));
+				motion_debris this_debris;
+				this_debris.name = "Nebula";
+
+				int count = 0;
+
+				while (optional_string("$DebrisNeb:")) {
 				in_check = true;
 
-				stuff_string(name, F_NAME, MAX_FILENAME_LEN);
+					stuff_string(name, F_NAME, MAX_FILENAME_LEN);
 
-				if (Num_debris_nebula < MAX_DEBRIS_VCLIPS) {
-					strcpy_s(Debris_vclips_nebula[Num_debris_nebula++].name, name);
+					if (count < MAX_MOTION_DEBRIS_BITMAPS) {
+						strcpy_s(this_debris.bitmaps[count++].name, name);
+					} else {
+						Warning(LOCATION, "Could not load nebula motion debris '%s'; maximum of %d exceeded.", name, MAX_MOTION_DEBRIS_BITMAPS);
+					}
 				}
-				else {
-					Warning(LOCATION, "Could not load nebula motion debris '%s'; maximum of %d exceeded.", name, MAX_DEBRIS_VCLIPS);
+				if (count == MAX_MOTION_DEBRIS_BITMAPS) {
+					Motion_debris.push_back(this_debris);
+				} else {
+					error_display(0, "Not enough bitmaps defined for motion debris '%s'. Skipping!\n", this_debris.name.c_str());
+				}
+			}
+
+			CHECK_END();
+
+			// custom debris pieces
+			while (optional_string("$Motion Debris Name:")) {
+				in_check = true;
+
+				stuff_string(name, F_NAME, MAX_NAME_LEN);
+
+				// check that the name is unique
+				int check = get_motion_debris_by_name(name);
+
+				if (check >= 0) {
+					error_display(0, "Motion debris name %s is already used; All names must be unique. Skipping!\n", name);
+					skip_to_start_of_string_either("$Motion Debris Name:", "#end");
```

Comment:
This is modular so overwriting an existing one should be allowed right? I'm not asking you to implement this if so, as this isn't inherently bad to do, but I wanted to make sure it wasn't a mistake.

Reply:
This now allows replacing an existing motion debris define. The downside for now, without adding full +nocreate support, is that any old data that wasn't replaced will stick around. For now this isn't a problem because you really should be replacing all 4 bitmaps with each define. But if additional data is added to the table parsing here, then it'd probably be wise to implement +nocreate and full error checking at that time.

Reply:
actually, it's late and I'm tired and should have done this the first time around. Now it will correctly require all 4 bitmaps to be defined during new parsing and error otherwise. This way leftover data is not an issue. Replacing one or more bitmaps, if that's a desired feature, can be done later with +nocreate.